ffi/utf8proc.lua

```diff
@@ -30,14 +30,19 @@ function Utf8Proc.lowercase(str)
     return folded_str
 end
 
-function Utf8Proc.strlen(str)
-    local str_buf = ffi.cast("const utf8proc_uint8_t *", str)
-    local buffer = ffi.new("utf8proc_int32_t[1]")
+function Utf8Proc.count(str)
+    local str_p = ffi.cast("const utf8proc_uint8_t *", str)
+    local codepoint = ffi.new("utf8proc_int32_t[1]")
     local count = 0
     local pos = 0
-    while str_buf[pos] ~= 0 do
-        count = count + 1
-        pos = pos + libutf8proc.utf8proc_iterate(str_buf + pos, -1, buffer)
+    while str_p[pos] ~= 0 do
```

Comment:
I still don't like that as a loop condition ;p. (Lua strings can contain NULs).

Reply:
Yes, but if I use
`while bytes > 0 do` I get random results for count. (Maybe depending on what is in the memory)
```
xxx str_len	18	uft8proc.count	35	text	Wahre Mitternacht:
xxx str_len	13	uft8proc.count	16	text	Astronomisch:
xxx str_len	9	uft8proc.count	36	text	Nautisch:
xxx str_len	11	uft8proc.count	35	text	Bürgerlich:
xxx str_len	14	uft8proc.count	36	text	Sonnenaufgang:
xxx str_len	14	uft8proc.count	16	text	Wahrer Mittag:
xxx str_len	16	uft8proc.count	44	text	Sonnenuntergang:
xxx str_len	11	uft8proc.count	35	text	Bürgerlich:
xxx str_len	9	uft8proc.count	36	text	Nautisch:
xxx str_len	13	uft8proc.count	16	text	Astronomisch:
xxx str_len	18	uft8proc.count	35	text	Wahre Mitternacht:
```

Correct results are calculated if I change the function to:
```
function Utf8Proc.count(str)
    local str_p = ffi.cast("const utf8proc_uint8_t *", str)
    local codepoint = ffi.new("utf8proc_int32_t[1]")
    local count = 0
    local pos = 0
    local bytes = 1
    local str_len = str:len()
    while pos < str_len and bytes > 0 do
        bytes = libutf8proc.utf8proc_iterate(str_p + pos, -1, codepoint)
        if bytes > 0 then
            count = count + 1
            pos = pos + bytes
        else
            return count, false
        end
    end
    return count
end
```
But this would mean to iterate to the string twice (str:len() and our loop). 

Would it be any good to use `while codepoint[0] ~= 0 and bytes > 0 do` or is this just the same as `str_p[pos] ~=0`?